### PR TITLE
[Fix](Variant) use uinque id to access column reader

### PR DESCRIPTION
**Pull request by eldenmoon**
Currently, the variant type is not supported rename column because its column reader accesses columns by path rather than by unique ID. If the name is modified, the column reader may not locate the column correctly.So we should access by unique id 

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/hierarchical_data_reader.cpp [hunk 1]
```diff
@@ -34,14 +36,15 @@ namespace segment_v2 {
 Status HierarchicalDataReader::create(std::unique_ptr<ColumnIterator>* reader,
                                       vectorized::PathInData path,
                                       const SubcolumnColumnReaders::Node* node,
-                                      const SubcolumnColumnReaders::Node* root) {
+                                      const SubcolumnColumnReaders::Node* root,
+                                      const vectorized::DataTypePtr& root_type) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/hierarchical_data_reader.cpp:40]: root_type is not used

File: be/src/olap/field.h [hunk 2]
```diff
@@ -28,6 +28,7 @@
 #include "olap/types.h"
 #include "olap/utils.h"
 #include "runtime/collection_value.h"
+#include "runtime/descriptors.h"
```
Comment (xiaokang) [inline be/src/olap/field.h:31]: unused

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/hierarchical_data_reader.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/olap/rowset/segment_v2/hierarchical_data_reader.cpp]: done

File: be/src/olap/field.h (hunk 2 above)
Comment (eldenmoon) [reply be/src/olap/field.h]: done

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

**Review by qidaye — APPROVED** [review]

**Review by amorynan — APPROVED** [review]
LGTM

**Pull request merged**